src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -74,6 +71,7 @@ internal partial class Http2Connection : IHttp2StreamLifetimeHandler, IHttpStrea
     internal readonly Http2KeepAlive? _keepAlive;
     internal readonly Dictionary<int, Http2Stream> _streams = new Dictionary<int, Http2Stream>();
     internal PooledStreamStack<Http2Stream> StreamPool;
+    internal Action<Http2Stream>? _onStreamCompleted;
```

Comment:
Is this just for testing?

I don't think it needs to be fixed now, but I'd like to see this reworked to utilize `IHttp2StreamLifetimeHandler`. Tests should support customizing the handler so it can be used to capture this event. I did this in the HTTP/3 tests and it works well.

Reply:
Maybe create a follow-up issue.